### PR TITLE
chore: Includes build in each PR check

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Includes build in each PR check.

Link to any related issue(s): CLOUDP-277647

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -8,12 +8,11 @@ on:
       - main
   pull_request:
 jobs:
-  golangci-lint:
+  code-health:
```
Comment (lantoli) [inline .github/workflows/pr.yml:11]: simplifying and joining linter and tests in the same job as their initialization is the same and it's what takes most of the time

File: .github/workflows/tools.yaml
Comment (lantoli) [inline .github/workflows/tools.yaml:1]: these checks are passed now in every PR

File: .github/workflows/pr.yml (hunk 1 above)
Comment (wtrocki) [reply .github/workflows/pr.yml:11]: I think someone had strong opinion about small jobs but forgot who had that :D
I'm ok with larger PR job.

Comment (lantoli) [reply .github/workflows/pr.yml:11]: i agree with that general idea, but in this case they have exactly the same initialization and second one depends on first one, so I thought they could be together, most of the code here is for initiazalition (which is duplicated in each former job)

File: tools/config/go-templates/client.mustache [hunk 2]
```diff
@@ -481,15 +481,15 @@ func (c *APIClient) makeApiError(res *http.Response, httpMethod, httpPath string
 
 	localVarBody, err := io.ReadAll(res.Body)
 	if err != nil {
-		newErr.error = err.Error()
+		newErr.error = fmt.Sprintf("(%s) failed to read response body: %s", res.Status, err.Error())
```
[the remark this answers is not in the document]
Comment (wtrocki) [reply tools/config/go-templates/client.mustache:484]: Thank you for doing it.

File: .github/workflows/pr.yml (hunk 1 above)
Comment (lantoli) [reply .github/workflows/pr.yml:11]: but happy to revert if you prefer that I leave them as they were

**Review by wtrocki — APPROVED** [review]
LGTM based on assumption that you have tested both tools linting (affected here) and regeneration. Job seems to be fast enough. In future PRs that affect CI/CD we can add extra notes to docs.

File: .github/workflows/pr.yml (hunk 1 above)
Comment (wtrocki) [reply .github/workflows/pr.yml:11]: I'm ok with the larger job

File: tools/config/go-templates/client.mustache (hunk 2 above)
[the remark this answers is not in the document]
Comment (EspenAlbert) [reply tools/config/go-templates/client.mustache:484]: Thank you!
I didn't know about the templates, but I will remember next time 👍

[the remark this answers is not in the document]
Comment (lantoli) [reply tools/config/go-templates/client.mustache:484]: @EspenAlbert no need to remember, PR check will fail next time :-)

**Pull request merged**